File: FinanceBackEnd/FinanceApi/Domain/Models/CurrencyExchangeRate.cs
using FinanceApi.Core.SpecialTypes;
using FinanceApi.Domain.Models.Base;

namespace FinanceApi.Domain.Models;

public class CurrencyExchangeRate : Entity<Guid>
{
    public CurrencyExchangeRate()
        : base()
    {
    }

    public Guid BaseCurrencyId { get; set; }

    public Guid QuoteCurrencyId { get; set; }

    public Money BuyRate { get; set; } = 0m;

    public Money SellRate { get; set; } = 0m;

    public DateTime TimeStamp { get; set; }

    public Currency BaseCurrency { get; set; }

    public Currency QuoteCurrency { get; set; }
}

File: FinanceFrontEnd/FinanceApp/app/types/currencyExchangeRate.ts
export interface CurrencyData {
  id: string;
  name: string;
  shortName: string;
}

export interface CurrencyExchangeRateRecord {
  id: string;
  createdAt: string;
  updatedAt: string;
  timeStamp: string;
  deactivated: boolean;
  buyRate: number;
  sellRate: number;
  baseCurrencyId: string;
  quoteCurrencyId: string;
  baseCurrency: CurrencyData;
  quoteCurrency: CurrencyData;
}
